### PR TITLE
ZX Spectrum Multiface One improvements

### PR DESCRIPTION
**Pull request by TwistedTom**
Corrected nmi clearing
Added on board kempston port
Added disable switch and joystick disable jumper
Lots of notes

I'm not too familiar with the other models so made changes to original "One" model only for now.

### CODE REVIEW

File: src/devices/bus/spectrum/mface.cpp [hunk 1]
```diff
@@ -24,37 +97,77 @@ DEFINE_DEVICE_TYPE(SPECTRUM_MPRINT, spectrum_mprint_device, "spectrum_mprint", "
 //  INPUT_PORTS( mface )
 //-------------------------------------------------
 
-INPUT_PORTS_START(mface)
+INPUT_PORTS_START( mface )
 	PORT_START("BUTTON")
-	PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_BUTTON1) PORT_NAME("Multiface") PORT_CODE(KEYCODE_F12) PORT_CHANGED_MEMBER(DEVICE_SELF, spectrum_mface1_device, magic_button, 0)
+	PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_BUTTON1) PORT_NAME("Red Button") PORT_CODE(KEYCODE_BACKSPACE) PORT_CHANGED_MEMBER(DEVICE_SELF, spectrum_mface_base_device, magic_button, 0)
```
Comment (cuavas) [inline src/devices/bus/spectrum/mface.cpp]: Why the change to backspace here?

File: src/devices/bus/spectrum/mface.cpp [hunk 2]
```diff
@@ -191,13 +305,18 @@ void spectrum_mface1_device::pre_opcode_fetch(offs_t offset)
 uint8_t spectrum_mface1_device::iorq_r(offs_t offset)
 {
 	uint8_t data = m_exp->iorq_r(offset);
+	
+	if (ioport("CONFIG")->read() & 0x01)  // disable switch
```
Comment (cuavas) [inline src/devices/bus/spectrum/mface.cpp]: Please use a `required_ioport` for this - you shouldn't have tag map lookups in the hot path.

File: src/devices/bus/spectrum/mface.cpp [hunk 3]
```diff
@@ -24,37 +97,77 @@ DEFINE_DEVICE_TYPE(SPECTRUM_MPRINT, spectrum_mprint_device, "spectrum_mprint", "
...
 	PORT_START("BUTTON")
-	PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_BUTTON1) PORT_NAME("Multiface") PORT_CODE(KEYCODE_F12) PORT_CHANGED_MEMBER(DEVICE_SELF, spectrum_mface1_device, magic_button, 0)
+	PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_BUTTON1) PORT_NAME("Red Button") PORT_CODE(KEYCODE_BACKSPACE) PORT_CHANGED_MEMBER(DEVICE_SELF, spectrum_mface_base_device, magic_button, 0)
+INPUT_PORTS_END
+
+INPUT_PORTS_START( mface1 )
+	PORT_INCLUDE( mface )
+	
+	PORT_START("CONFIG")
+	PORT_CONFNAME(0x01, 0x00, "Disable Switch")
```
Comment (cuavas) [inline src/devices/bus/spectrum/mface.cpp]: Right now if the user hits the "magic button" and then quickly flips this off before NMI is cleared, NMI will be stuck on.  You need to either always clear NMI when the acknowledge address it read, or add a port changed member on this switch to drop NMI if the interface is disabled while NMI is asserted.

File: src/devices/bus/spectrum/mface.cpp (hunk 1 above)
Comment (DavidHaywood) [reply src/devices/bus/spectrum/mface.cpp]: it F12 conflicts with debugger use, which is quite a major problem, I spent ages trying to chase down another issue until I realised F12, to exit debugger, was breaking things.

Comment (TwistedTom) [reply src/devices/bus/spectrum/mface.cpp]: Yes it was both that and simply that F12 is "snapshot" button by default in Mame, backspace seems to have no default mapping.

File: src/devices/bus/spectrum/mface.cpp (hunk 2 above)
Comment (TwistedTom) [reply src/devices/bus/spectrum/mface.cpp]: Ok I will change that...

File: src/devices/bus/spectrum/mface.cpp (hunk 3 above)
Comment (TwistedTom) [reply src/devices/bus/spectrum/mface.cpp]: I was aware of this but I think it's actually the correct behaviour of the h/w. The multiface osd doesn't clear the nmi until it is closed with the 'r' key. So if it's on screen and the disable switch it opened then the nmi is never cleared. The circuit is a little odd, the "magic button" triggers a flip-flop that immediately disables the button, the disable switch breaks a signal made up of /rd and /iorq lines, I think the original designers just assumed you wouldn't do that (why would you?). I think I'll just remove the disable switch for now until I can confirm exact behaviour on real h/w.

Comment (cuavas) [reply src/devices/bus/spectrum/mface.cpp]: Oh, if the real hardware actually does let the NMI get stuck like that, just add a note about it.  I wouldn't have expected the real hardware to do that - I would have thought it would do the "safe" thing and either still allow the NMI to be disabled or drop it on being disabled.
